simtools/simtel/simtel_histograms.py

```diff
@@ -79,17 +75,17 @@ def list_of_histograms(self):
         list:
             List of histograms.
         """
-
-        self._list_of_histograms = []
-        for file in self._histogram_files:
-            with EventIOFile(file) as f:
-                for o in yield_toplevel_of_type(f, Histograms):
-                    try:
-                        hists = o.parse()
-                    except Exception:  # pylint: disable=broad-except
-                        self._logger.warning(f"Problematic file {file}")
-                        continue
-                    self._list_of_histograms.append(hists)
+        if self._list_of_histograms is None:
+            self._list_of_histograms = []
+            for file in self._histogram_files:
+                with EventIOFile(file) as f:
+                    for o in yield_toplevel_of_type(f, Histograms):
+                        try:
+                            hists = o.parse()
+                        except Exception:  # pylint: disable=broad-except
+                            self._logger.warning(f"Problematic file {file}")
```

Comment:
I don't know about the severity of this, but maybe raising an error here would be better?

Reply:
I was also wondering about it (did not add those lines now). But I could not come up with a test for this. I guess we would have to mess up the simtel output file such that the parser from eventio does not work. That is not supposed to be done in simtools though. Not raising an error above means we want to access the file even if the parser does not work. If we know what we are doing in the following, I think that is fine.